cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -73,129 +80,99 @@ def transfer_flow_cell(
-            return False
+        for metric in sample_lane_sequencing_metrics:
+            if not self.metric_exists_in_status_db(metric):
+                self.add_metric_to_status_db(metric)
+        self.status_db.session.commit()
 
-        return True
+    def metric_exists_in_status_db(self, metric: SampleLaneSequencingMetrics) -> bool:
+        existing_metrics_entry = (
+            self.status_db.get_metrics_entry_by_flow_cell_name_sample_internal_id_and_lane(
+                flow_cell_name=metric.flow_cell_name,
+                sample_internal_id=metric.sample_internal_id,
+                lane=metric.flow_cell_lane_number,
+            )
+        )
+        if existing_metrics_entry:
+            LOG.warning(
+                f"Sample lane sequencing metrics already exist for {metric.flow_cell_name}, {metric.sample_internal_id}, and {metric.flow_cell_lane_number}. Skipping."
+            )
+        return bool(existing_metrics_entry)
 
-    def is_demultiplexing_complete(self, flow_cell_directory: Path) -> bool:
-        return Path(flow_cell_directory, DemultiplexingDirsAndFiles.DEMUX_COMPLETE).exists()
+    def add_metric_to_status_db(self, metric: SampleLaneSequencingMetrics) -> None:
+        LOG.debug(
+            f"Adding sample lane sequencing metrics for {metric.flow_cell_name}, {metric.sample_internal_id}, and {metric.flow_cell_lane_number}."
+        )
+        self.status_db.session.add(metric)
 
     def update_sample_read_counts(self, flow_cell_data: FlowCellDirectoryData) -> None:
         """Update samples in status db with the sum of all read counts for the sample in the sequencing metrics table."""
 
-        q30_threshold: int = self.get_q30_threshold(flow_cell_data.sequencer_type)
-
-        sample_internal_ids: List[str] = self.get_sample_ids_from_sample_sheet(
-            flow_cell_data=flow_cell_data
-        )
+        q30_threshold: int = get_q30_threshold(flow_cell_data.sequencer_type)
+        sample_internal_ids: List[str] = get_sample_ids_from_sample_sheet(flow_cell_data)
 
         for sample_id in sample_internal_ids:
-            self.update_single_sample_read_count(sample_id=sample_id, q30_threshold=q30_threshold)
-        self.status_db.session.commit()
+            self.update_sample_read_count(sample_id=sample_id, q30_threshold=q30_threshold)
 
-    def get_q30_threshold(self, sequencer_type: Sequencers) -> int:
-        return FLOWCELL_Q30_THRESHOLD[sequencer_type]
+        self.status_db.session.commit()
 
-    def update_single_sample_read_count(self, sample_id: str, q30_threshold: int) -> None:
+    def update_sample_read_count(self, sample_id: str, q30_threshold: int) -> None:
         """Update the read count for a sample in status db with all reads exceeding the q30 threshold from the sequencing metrics table."""
         sample = self.status_db.get_sample_by_internal_id(internal_id=sample_id)
 
```

Comment:
Do we need to do the `if sample:` in the line below here? In what use case would we not have a sample to add the read counts to? and if we don't - would we not want to be somewhat informed? I think logging with a warning `LOG.warning(f"Cannot find {sample_id} in status_db when adding read counts")` - this or remove the if and let it fail 🤷

Reply:
Hm, my best guess is that it could occur due to some issue with the porting of samples from cgstats to cg. But I think @ChrOertlin is better informed - do you think this is something we should handle?

Reply:
I guess we have to since the return type of the `get_sample_by_internal_id` function is an optional Sample

Reply:
Added the log warning

Reply:
nice!